oak_runtime/src/introspect.rs

```diff
@@ -25,6 +25,71 @@ use prost::Message;
 use regex::Regex;
 use std::{net::SocketAddr, sync::Arc};
 
+#[cfg(not(feature = "oak_introspection_client"))]
+mod introspection_client {
+    use super::*;
+
+    pub fn find_client_file(_path: &str) -> Option<Response<Body>> {
+        None
+    }
+}
+
+#[cfg(feature = "oak_introspection_client")]
+mod introspection_client {
+    use super::*;
+    use hyper::header::CONTENT_ENCODING;
+
+    fn static_file(
+        file: &str,
+        content_type: &str,
+        content_encoding: Option<&str>,
+    ) -> Response<Body> {
+        let mut response = Response::new(Body::from(file.to_string()));
+        let headers = response.headers_mut();
+        headers.insert(CONTENT_TYPE, content_type.parse().unwrap());
+        if let Some(content_encoding_value) = content_encoding {
+            headers.insert(CONTENT_ENCODING, content_encoding_value.parse().unwrap());
+        }
+
+        response
+    }
+
+    // Looks for a matching file used by the browser client and returns it
+    pub fn find_client_file(path: &str) -> Option<Response<Body>> {
+        let subpath = Regex::new(r"^/dynamic(?P<filepath>[^\s]*)$")
+            .unwrap()
+            .captures(path)?
+            .name("filepath")
+            .unwrap()
+            .as_str();
+
+        match subpath {
+            "/index.js" => Some(static_file(
+                "../introspection_browser_client/dist/index.js.gz",
+                "application/javascript",
+                Some("gzip"),
+            )),
+            "/graphvizlib.wasm" => Some(static_file(
+                "../introspection_browser_client/dist/graphvizlib.wasm.gz",
+                "application/wasm",
+                Some("gzip"),
+            )),
+            "/favicon.png" => Some(static_file(
+                "../introspection_browser_client/dist/favicon.png",
+                "image/png",
+                None,
```

Comment:
Can we just gzip everything?

Reply:
I did consider it, but it would reduce code complexity slightly. But realistically it only really makes sense for text based assets. For images etc it only adds CPU overhead while potentially even increasing file size. 

I'm personally ok with a few more lines of code if it means we follow web-dev best practices.